### PR TITLE
Add missing argument names in function definitions

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/mix/lib/mix/tasks/archive.install.ex [hunk 1]
```diff
@@ -62,8 +62,7 @@ defmodule Mix.Tasks.Archive.Install do
   end
 
   # Callbacks
-
-  def check_install_spec({local_or_url, path_or_url}, _opts) when
+  def check_install_spec({local_or_url, path_or_url} = _install_inspect, _opts) when
```
Comment (whatyouhide) [inline lib/mix/lib/mix/tasks/archive.install.ex]: Is this `_install_spec` instead of `_install_inspect`?

File: lib/elixir/lib/map_set.ex [hunk 2]
```diff
@@ -347,6 +347,8 @@ defmodule MapSet do
 
   """
   @spec union(t(val1), t(val2)) :: t(val1 | val2) when val1: value, val2: value
+  def union(map_set1, map_set2)
```
Comment (whatyouhide) [inline lib/elixir/lib/map_set.ex]: We strictly don't need this so I'm not 100% sure it's useful to have it. 🤔

File: lib/mix/lib/mix/tasks/archive.install.ex (hunk 1 above)
Comment (eksperimental) [inline lib/mix/lib/mix/tasks/archive.install.ex]: oops

File: lib/elixir/lib/map_set.ex (hunk 2 above)
Comment (eksperimental) [inline lib/elixir/lib/map_set.ex]: Well, you do cause otherwise your signature looks like "union(map_set, arg2)"

File: lib/mix/lib/mix/tasks/archive.install.ex (hunk 1 above)
Comment (eksperimental) [inline lib/mix/lib/mix/tasks/archive.install.ex]: fixed

File: lib/elixir/lib/map_set.ex (hunk 2 above)
Comment (whatyouhide) [inline lib/elixir/lib/map_set.ex]: I was looking at the docs for 1.4.4 but this changed apparently so you're right :)

Comment (josevalim) [inline lib/elixir/lib/map_set.ex]: Please open up a bug report because I don't see why it changed to `arg2`.

Comment (josevalim) [inline lib/elixir/lib/map_set.ex]: I.e. let's undo this change and do a bug report so the current version works™.

Comment (eksperimental) [inline lib/elixir/lib/map_set.ex]: what is the last version know to have worked?

**Comment by josevalim** [comment]
V1.4 branch.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

File: lib/elixir/lib/map_set.ex (hunk 2 above)
Comment (eksperimental) [inline lib/elixir/lib/map_set.ex]: ok, I found the offending commit doing a git bisect, 
https://github.com/elixir-lang/elixir/commit/bc9df398759769a7656589b51e32809071e8df7c#diff-0a5412a70e42a886648e41d22f7d17aeR350

the good thing is that the fix that I have for https://github.com/elixir-lang/elixir/pull/6195 solves this.
but instead of doing:
`def union(%MapSet{map: map1, version: version} = map_set, %MapSet{map: map2, version: version}) do`
we will have to do
`def union(%MapSet{map: map1, version: version} = map_set1, %MapSet{map: map2, version: version} = _map_set2) do`
otherwise the signature will be `union(map_set, map_set)`
and we want `union(map_set1, map_set2)`
So i think it is better to introduce my change.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
